### PR TITLE
Added http HealthIndicator

### PR DESCRIPTION
**Pull request by HarryEMartland**
I have added a new Health Indicator and Auto Configuration for it.
This indicator makes a get request to a url reporting if the response is ok.
I keep writing and seeing people write similar things to this over and over again at work.

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/http/HttpHealthIndicator.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	 * @param url the url to be checked
+	 */
+	public HttpHealthIndicator(HttpClient httpClient, String url) {
+		this.httpClient = httpClient;
+		this.url = url;
+	}
+
+	@Override
+	protected void doHealthCheck(Health.Builder builder) {
+		HttpGet httpGet = new HttpGet(this.url);
```
Comment (spencergibb) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/http/HttpHealthIndicator.java]: HttpClient is optional, might be better to use RestTemplate.

**Comment by snicoll** [comment]
Thanks for the PR. A `HealthIndicator` is focused on a component and not a "technical mean to check if a service is up". As such, I am not keen to introduce a generic http health indicator as it does not fit with the general idea of an health indicator.

Let's see what the rest of the team thinks.

**Comment by bclozel** [comment]
I'm not in favor of doing this for the reasons @snicoll mentioned. In this case, the application should use the client library or the actual HTTP client (with credentials?) being used to check that the component is working - which is much more than just checking that the remote HTTP endpoint is up.

There are a lot of subtle limitations and configuration aspects to it - which explains why it should be a custom indicator in the first place.

@HarryEMartland also as a side note, if you're using this indicator in your own project - the configured timeout won't work if the remote endpoint is HTTPS or if you need to go through a proxy (see #11379).

**Comment by wilkinsona** [comment]
I agree with @snicoll and @bclozel.

**Comment by snicoll** [comment]
Thanks all.

@HarryEMartland, we won't merge the PR for the reasons described above. Thanks anyway!

**Pull request closed without merging**